BaseCollector.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from abc import ABC, abstractmethod
+from os import getenv
+
+import modules.TimedBlacklist as blacklist
+from modules.NetboxHelper import NetboxHelper
+from modules.VCenterConnection import VCenterConnection
+
+logger = logging.getLogger('esxi-exporter')
+
+
+class BaseCollector(ABC):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.vCenter = VCenterConnection(getenv('VCENTER_URL'), getenv('VCENTER_USER'), getenv('VCENTER_PASSWORD'))
+        self.netbox = NetboxHelper()
+

```

Comment:
please add a describe function.
All collectors should have it too. This avoids an initial run of `collect()` from prometheus to figure out respective metrics. 
Ref: last paragraph of https://github.com/prometheus/client_python#custom-collectors

Reply:
done.

Reply:
done.